.github/workflows/action-e2e.yaml

```diff
@@ -0,0 +1,147 @@
+        with:
+          path: this
+      - name: Setup test fixture
+        run: |
+          set -o errexit -o nounset -o pipefail -o xtrace
+
+          # Create a release archive from a fixture
+          tar -czvf archive.tar.gz -C this/e2e/fixtures/versioned --transform 's,^./,versioned-1.0.0/,' --sort=name --owner=root:0 --group=root:0 --mtime='UTC 1980-02-01' .
+
+          # Substitute the archive url to a local file path
+          cat this/e2e/fixtures/versioned/.bcr/source.template.json | jq ".url = \"file://$(realpath archive.tar.gz)\"" > "/tmp/source.template.json"
+          mv /tmp/source.template.json this/e2e/fixtures/versioned/.bcr/source.template.json
+          cat this/e2e/fixtures/versioned/.bcr/source.template.json
+      - name: Create registry
+        run: |
+          mkdir -p bazel-central-registry/modules
+          cd bazel-central-registry
+          git init
+      - name: Create entry
+        uses: ./this
+        with:
+          tag: v1.0.0
+          module-version: 1.0.0
+          github-repository: foobar/versioned
+          templates-dir: this/e2e/fixtures/versioned/.bcr
+          local-registry: bazel-central-registry
+      - name: Test entry content
+        run: |-
+          set -o errexit -o nounset -o pipefail -o xtrace
+
+          mkdir -p expected/1.0.0
+
+          cat > expected/metadata.json <<- EOF
+          {
+              "homepage": "https://github.com/testorg/versioned",
+              "maintainers": [
+                  {
+                      "name": "Foo McBar",
+                      "email": "foo@test.org",
+                      "github": "foobar"
+                  }
+              ],
+              "repository": [
+                  "github:testorg/versioned"
+              ],
+              "versions": [
+                  "1.0.0"
+              ],
+              "yanked_versions": {}
+          }
+          EOF
+          cat > expected/1.0.0/MODULE.bazel <<- EOF
+          module(
+              name = "versioned",
+              version = "1.0.0",
+          )
+          EOF
+          cat > expected/1.0.0/source.json <<- EOF
+          {
+              "integrity": "sha256-t39EUhiPjM65oWhYtLQ2dq4fWDU/oQ0L/DNBvYkNfyk=",
+              "strip_prefix": "versioned-1.0.0",
+              "url": "file:///home/runner/work/publish-to-bcr/publish-to-bcr/archive.tar.gz"
+          }
+          EOF
+          cat > expected/1.0.0/presubmit.yml <<- EOF
+          bcr_test_module:
+            module_path: "e2e/bzlmod"
+            matrix:
+              platform: ["debian10", "macos", "ubuntu2004", "windows"]
+              bazel: [6.x, 7.x]
+            tasks:
+              run_tests:
+                name: "Run test module"
+                platform: ${{ '\${{ platform }}' }}
+                bazel: ${{ '\${{ bazel }}' }}
+                test_targets:
+                  - "//..."
+          EOF
+
+          diff -r bazel-central-registry/modules/versioned expected
```

Comment:
nit: can we use `jq` to compare json so it's less brittle on things like differing ordering of keys?